### PR TITLE
Implement data processing methods

### PR DESCRIPTION
**Pull request by alexeymatyas**
@enyalux@gmail.com

### CODE REVIEW

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 1]
```diff
@@ -51,7 +70,20 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition7(){
-        return null;
+        return getCountBy(RoadAccident::getRoadSurfaceConditions);
```
Comment (bug-hunterx) [inline src/main/java/com/epam/processor/DataProcessor.java:73]: Technically it's not J7 :) 
I think this syntax is from J8...

But yea - it's fine

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 2]
```diff
@@ -93,31 +151,53 @@ public RoadAccident getAccidentByIndex(String index){
...
     /**
      * To match streaming operations result, return type is a java collection instead of multimap
      * @return
      */
     public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
+        return roadAccidentList.stream()
+                .collect(groupingBy(
+                        RoadAccident::getDistrictAuthority,
+                        HashMap::new,
```
Comment (bug-hunterx) [inline src/main/java/com/epam/processor/DataProcessor.java:198]: That's perfect solution from the first submition - very good. 
FYI - I did by myself without add HashMap::new - works fine as well.

File: src/main/java/com/epam/processor/DataProcessor.java (hunk 1 above)
Comment (alexeymatyas) [inline src/main/java/com/epam/processor/DataProcessor.java:73]: Oops:)

File: src/main/java/com/epam/processor/DataProcessor.java (hunk 2 above)
Comment (alexeymatyas) [inline src/main/java/com/epam/processor/DataProcessor.java:198]: Thanks! I didn't notice such overloaded method.

**Comment by bug-hunterx** [comment]
Closing as homework is accepted

**Pull request closed without merging**